src/Sensor/SensorController.java

```diff
@@ -1,55 +1,102 @@
 package Sensor;
 
 
-
-
 import org.hyperic.sigar.Sigar;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
 
-import java.util.ArrayList;
-import java.util.List;
 /**
  *
  * Created by jak12 on 30.04.2017.
  */
 public class SensorController {
 
+    //json mapping
+    private final static String DATATYPE_KEY = "datatype";
+    private final static String RESOURCE_ID_KEY = "resourceID";
+    private final static String DATA_VALUE = "data";
+    private final static String METADATA_VALUE = "metadata";
+    private final static String NAME_KEY = "name";
+    private final static String DESCRIPTION_KEY = "description";
+    private final static String MEASURE_TYPE_KEY = "measureType";
+    private final static String MEASURE_UNIT_KEY = "unit";
+    private final static String MEASURE_ID_KEY = "measureID";
+    private final static String MAX_VALUE_KEY = "maxValue";
+    private final static String MEASURESARRAY_KEY = "measuresArray";
+    private final static String MEASURE_VALUE_KEY = "value";
+    private final static String TIMESTAMP_KEY = "timestamp";
+
+    private final static String DATE_PATTERN = "yyyy-MM-dd'T'HH:mm:ss";
+
+    private JSONObject dataForMonitor = new JSONObject();
+    private JSONArray measuresJsonArray = new JSONArray();
+    private String resourceID = "";
+    private String hostname = "";
+    private String hostDescription = "";
+
+
+    private UDPConnectionController udpController = new UDPConnectionController();
     private List<Measurement> measurementsList = new ArrayList<>();
-    private String data;
 
-    public void startSensor(){
-        if(registerSensor()) {
-            chooseMeasurements();
-            sendMetadata();
+    private Sigar sigar;
 
-            //noinspection InfiniteLoopStatement
-            while(true) run();
+    public void startSensor(){
+        configureSensor();
 
+        //noinspection InfiniteLoopStatement
+        while(true) {
+            sendMetadata();
+            try {
+                Thread.sleep(1000);
```

Comment:
I don't really get this part, consult it with me. Also, the same try / catch is somewhere else, you (probably) could extract it to some global utility class